Jenkinsfile.cloud

```diff
@@ -39,6 +39,23 @@ node(env.NODE) {
         }
     }
 
+    // Check if there's a new version out.
+    // Really, we should be checksumming e.g. the ks and tdl too.
+    def latest_version, last_build_version
+    stage("Check for Changes") {
+        latest_version = sh(returnStdout: true,
+                            script: "ostree show --repo=${repo} --print-metadata-key=version ${ref}").trim()
+        latest_version = latest_version[1..-2] // trim single quotes
+        last_build_version = sh(returnStdout: true,
+                                script: "readlink ${output}/cloud/latest").trim()
+    }
+
+    if (latest_version == last_build_version) {
```

Comment:
What I've done elsewhere is have a "blow the cache" key in metadata.  If Pagure was up, I'd link to what https://pagure.io/fedora-atomic-host-continuous is doing, but basically: Let's have a yaml file like:

`build-caching.yml`:
```yaml
cloud-force-rebuild: 3.10-7
````

Then we add a condition here like `if (latest_version != cloud_force_rebuild && latest_version == last_build_version` or so, where we parse `cloud_force_rebuild` from the yaml in the job.